### ISSUE TITLE
Library scan stops quickly, very little media processed.

### ISSUE THREAD

**Issue by jacobmetrick** [jellyfin/jellyfin#2092]
**Describe the bug**
When I try to scan or rescan media in my library, very little media is processed and it cancels quite quickly.

**To Reproduce**
1. Scan a library with some video/music files in it.

**Expected behavior**
I expect the library scan to take some time and scan multiple files.

**Logs**
Here are the relevant logs, I believe:

```
[2019-12-02 08:27:01.470 -06:00] [INF] Executing "Scan media library"
[2019-12-02 08:27:01.478 -06:00] [INF] Stopping directory watching for path "/mnt/array/data/Movies"
[2019-12-02 08:27:01.487 -06:00] [INF] Validating media library
[2019-12-02 08:27:04.893 -06:00] [INF] MovieDbProvider: Finding id for item: Back to the Future Part II
[2019-12-02 08:28:45.167 -06:00] [INF] "Scan media library" Cancelled after 1 minute(s) and 43 seconds
[2019-12-02 08:28:45.168 -06:00] [INF] ExecuteQueuedTasks
[2019-12-02 08:28:45.204 -06:00] [INF] Watching directory /mnt/array/data/Movies
```

If there's a way to enable more logging, I'd do it, but I'm not currently sure how.

**System (please complete the following information):**
 - OS: Linux (Arch)
 - Browser: Firefox
 - Jellyfin Version: 10.4.1
 - Reverse proxy: nginx, but issue also exists with none

**Comment by jacobmetrick** [comment]
Still repros on 10.4.2, fwiw.

I wonder what's causing the cancellation after a minute or two?

**Comment by C-Fu** [comment]
Same here in my case. scanning a large library (like TV Shows) ended quite quickly, and it won't scrape all of them. Had to do it again and again.

If there's a way to automate library scanning (say, every 5 minutes) then this solves the issue somewhat.

**Comment by jacobmetrick** [comment]
Was able to repro this issue again in 10.4.3 as well as in master branch. All my tasks time out after 1 minute 40 seconds, roughly (aka 100 seconds). I assume this timeout is specified somewhere, but I haven't found exactly where yet.

Playing around with master a bit I was able to find an interesting stacktrace during scanning that comes up almost instantaneously; despite being labeled `TaskCanceledException` the task gets cancelled after the 100 second timeout, officially.
```
System.Threading.Tasks.TaskCanceledException: The operation was canceled.
   at System.Net.Http.ConnectHelper.ConnectAsync(String host, Int32 port, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.ConnectAsync(HttpRequestMessage request, Boolean allowHttp2, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.CreateHttp11ConnectionAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.GetHttpConnectionAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.SendWithRetryAsync(HttpRequestMessage request, Boolean doRequestAuth, CancellationToken cancellationToken)
   at System.Net.Http.RedirectHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpClient.FinishSendAsyncUnbuffered(Task`1 sendTask, HttpRequestMessage request, CancellationTokenSource cts, Boolean disposeCts)
   at Emby.Server.Implementations.HttpClientManager.HttpClientManager.SendAsyncInternal(HttpRequestOptions options, HttpMethod httpMethod) in /home/jacob/Development/jellyfin/Emby.Server.Implementations/HttpClientManager/HttpClientManager.cs:line 265
   at Emby.Server.Implementations.HttpClientManager.HttpClientManager.SendAsync(HttpRequestOptions options, HttpMethod httpMethod) in /home/jacob/Development/jellyfin/Emby.Server.Implementations/HttpClientManager/HttpClientManager.cs:line 171
   at MediaBrowser.Providers.Manager.ProviderManager.SaveImage(BaseItem item, String url, ImageType type, Nullable`1 imageIndex, CancellationToken cancellationToken) in /home/jacob/Development/jellyfin/MediaBrowser.Providers/Manager/ProviderManager.cs:line 151
   at Emby.Server.Implementations.Library.LibraryManager.ConvertImageToLocal(BaseItem item, ItemImageInfo image, Int32 imageIndex) in /home/jacob/Development/jellyfin/Emby.Server.Implementations/Library/LibraryManager.cs:line 2825
   at Emby.Drawing.ImageProcessor.ProcessImage(ImageProcessingOptions options) in /home/jacob/Development/jellyfin/Emby.Drawing/ImageProcessor.cs:line 163
   at MediaBrowser.Api.Images.ImageService.GetImageResult(BaseItem item, Guid itemId, ImageRequest request, ItemImageInfo image, Boolean cropwhitespace, IReadOnlyCollection`1 supportedFormats, IReadOnlyCollection`1 enhancers, Nullable`1 cacheDuration, IDictionary`2 headers, Boolean isHeadRequest) in /home/jacob/Development/jellyfin/MediaBrowser.Api/Images/ImageService.cs:line 646
   at Emby.Server.Implementations.Services.ServiceExecGeneral.GetTaskResult(Task task) in /home/jacob/Development/jellyfin/Emby.Server.Implementations/Services/ServiceExec.cs:line 108
   at Emby.Server.Implementations.Services.ServiceHandler.ProcessRequestAsync(HttpListenerHost httpHost, IRequest httpReq, HttpResponse httpRes, ILogger logger, CancellationToken cancellationToken) in /home/jacob/Development/jellyfin/Emby.Server.Implementations/Services/ServiceHandler.cs:line 80
   at Emby.Server.Implementations.HttpServer.HttpListenerHost.RequestHandler(IHttpRequest httpReq, String urlString, String host, String localPath, CancellationToken cancellationToken) in /home/jacob/Development/jellyfin/Emby.Server.Implementations/HttpServer/HttpListenerHost.cs:line 518
```

Given this I'd recommend reopening the bug as I believe the merged fix at #2104 made the 10.4.3 release, and I've reproed on both that version and on master, but this is of course up to the maintainers.

If I find anything else/a fix I'll be sure to submit, but I thought I'd come back with some knowledge I've gained.

**Comment by anthonylavado** [comment]
I’ll re-open, but in testing for scanning issues on macOS, Windows, and Debian, I would actually have scans that take 8-9 minutes, and succeed.

I don’t know what else could be happening, because there doesn’t seem to be a timeout that I know of.

**Comment by anthonylavado** [comment]
Also reading closer @jacobmetrick, it looks like your system is having trouble getting a metadata image from a provider? Perhaps there is an HTTP timeout that you’re running in to

**Comment by jacobmetrick** [comment]
Yeah, not sure personally exactly what is going on yet; I think the person in this reddit post may have been having the same issue, FWIW https://www.reddit.com/r/jellyfin/comments/cbiz3m/library_will_not_update_after_a_point_in_1053/ 

I think your theory seems promising about an HTTP timeout. It usually takes roughly 3 seconds to hit the stack I posted above, which seems like a probable length for an HTTP timeout. I'm not sure where that timeout would be strictly configured; I haven't run into anything in the jellyfin code that would suggest one explicitly, but it may be provided by the dotnet binary on my particular install.

On another note that may be helpful, I was wondering how to turn on debug logging. Fundamentally I've been running jellyfin by a command like `dotnet jellyfin.dll` but I haven't yet been able to figure out the right argument to turn on the debug logging. Statements like [these](https://github.com/jellyfin/jellyfin/blob/61b9845e0f72fc145a12dba38288d01baa6e03be/Emby.Server.Implementations/Library/LibraryManager.cs#L2823) appear very close to where my stack occurs and could be useful, but I've just not figured out how to turn them on. Some online documentation recommends running with `-v diag` as an argument but putting it before the dll in the execution line confuses dotnet and putting it after the dll has it interpreted as a command line argument to the dll itself. But maybe I'm going about it entirely wrong?

**Comment by Bond-009** [comment]
@jacobmetrick To enable debug logging: https://jellyfin.org/docs/general/administration/connectivity.html#debug-more-verbose-logging

**Comment by Bond-009** [comment]
Maybe this is related to this? https://github.com/dotnet/corefx/issues/20296

**Comment by jacobmetrick** [comment]
Took another little look at importing media; I was able to turn on debug logging (thanks @Bond-009 ) and I noticed I would generally have the import process hang on `Running OmdbItemProvider for...`. Couldn't really figure out exactly what was causing the issues with that provider, but disabling Open Movie Database as a metadata provider allowed way more files to be scanned than ever before.

One hint, perhaps, is that I will see the TmdbMovieProvider tend to make a couple ms backoff when it queries the third party metadata provider; I don't know if the Omdb version does that and just doesn't say or it doesn't, or even if it needs to.

Anyways, disabling OmdbItemProvider got me a lot further (from having ~25 files scanned to 938) but I do have 1312 files in this particular library, and right now it's hung up on Fantastic Four (2015). I checked TMDB manually and it seems like a valid entry, but the scan sort of stops at this point:

```
[07:46:14] [DBG] Running FolderImageProvider for /var/lib/jellyfin/root
[07:46:15] [DBG] Running FFProbeProvider for /mnt/array/data/Movies/Fantastic Four (2015)/Fantastic Four (2015).mkv
[07:46:15] [DBG] /usr/bin/ffprobe -i file:"/mnt/array/data/Movies/Fantastic Four (2015)/Fantastic Four (2015).mkv" -threads 0 -v warning -print_format json -show_streams -show_chapters -show_format
[07:46:15] [DBG] Starting ffprobe with args -i file:"/mnt/array/data/Movies/Fantastic Four (2015)/Fantastic Four (2015).mkv" -threads 0 -v warning -print_format json -show_streams -show_chapters -show_format
[07:46:16] [DBG] Running MovieNfoProvider for /mnt/array/data/Movies/Fantastic Four (2015)/Fantastic Four (2015).mkv
[07:46:16] [DBG] MovieNfoProvider returned no metadata for /mnt/array/data/Movies/Fantastic Four (2015)/Fantastic Four (2015).mkv
[07:46:16] [DBG] Running TmdbMovieProvider for /mnt/array/data/Movies/Fantastic Four (2015)/Fantastic Four (2015).mkv
```

**Comment by tyhawk** [comment]
I have the same issue and the scraper also stops at movies (all other media works fine:
```
[2020-01-06 12:05:30.637 +01:00] [DBG] Running "FFProbeProvider" for "/media/Speelfilm/Animatie/The Secret Life of Pets 2 (2019)/The Secret Life of Pets 2 (2019).mkv"
[2020-01-06 12:05:30.670 +01:00] [DBG] "/usr/local/bin/ffprobe" "-i file:\"/media/Speelfilm/Animatie/The Secret Life of Pets 2 (2019)/The Secret Life of Pets 2 (2019).mkv\" -threads 0 -v warning -print_format json -show_streams -show_chapters -show_format"
[2020-01-06 12:05:30.670 +01:00] [DBG] Starting ffprobe with args "-i file:\"/media/Speelfilm/Animatie/The Secret Life of Pets 2 (2019)/The Secret Life of Pets 2 (2019).mkv\" -threads 0 -v warning -print_format json -show_streams -show_chapters -show_format"
[2020-01-06 12:05:32.949 +01:00] [DBG] Running "MovieNfoProvider" for "/media/Speelfilm/Animatie/The Secret Life of Pets 2 (2019)/The Secret Life of Pets 2 (2019).mkv"
[2020-01-06 12:05:32.952 +01:00] [DBG] "MovieNfoProvider" returned no metadata for "/media/Speelfilm/Animatie/The Secret Life of Pets 2 (2019)/The Secret Life of Pets 2 (2019).mkv"
[2020-01-06 12:05:32.952 +01:00] [DBG] Running "TmdbMovieProvider" for "/media/Speelfilm/Animatie/The Secret Life of Pets 2 (2019)/The Secret Life of Pets 2 (2019).mkv"
[2020-01-06 12:07:13.102 +01:00] [INF] "Scan media library" Cancelled after 3 minute(s) and 45 seconds
```
I tried scanning the same subfolder without the Open Movie DB (using only themoviedb as a source for meta data):
```
[2020-01-06 12:17:15.669 +01:00] [DBG] Running "FFProbeProvider" for "/media/Speelfilm/Animatie/The Secret Life of Pets 2 (2019)/The Secret Life of Pets 2 (2019).mkv"
[2020-01-06 12:17:15.669 +01:00] [DBG] "/usr/local/bin/ffprobe" "-i file:\"/media/Speelfilm/Animatie/The Secret Life of Pets 2 (2019)/The Secret Life of Pets 2 (2019).mkv\" -threads 0 -v warning -print_format json -show_streams -show_chapters -show_format"
[2020-01-06 12:17:15.669 +01:00] [DBG] Starting ffprobe with args "-i file:\"/media/Speelfilm/Animatie/The Secret Life of Pets 2 (2019)/The Secret Life of Pets 2 (2019).mkv\" -threads 0 -v warning -print_format json -show_streams -show_chapters -show_format"
[2020-01-06 12:17:16.283 +01:00] [DBG] Running "MovieNfoProvider" for "/media/Speelfilm/Animatie/The Secret Life of Pets 2 (2019)/The Secret Life of Pets 2 (2019).mkv"
[2020-01-06 12:17:16.285 +01:00] [DBG] "MovieNfoProvider" returned no metadata for "/media/Speelfilm/Animatie/The Secret Life of Pets 2 (2019)/The Secret Life of Pets 2 (2019).mkv"
[2020-01-06 12:17:16.285 +01:00] [DBG] Running "TmdbMovieProvider" for "/media/Speelfilm/Animatie/The Secret Life of Pets 2 (2019)/The Secret Life of Pets 2 (2019).mkv"
[2020-01-06 12:18:56.372 +01:00] [INF] "Scan media library" Cancelled after 2 minute(s) and 58 seconds
```
As I had problems scraping TV shows and in the end the issue was my file naming convention, I also tried to follow recommendations in the docs for movie files (and excluding the Open Movie DB as a source for meta data), but this yielded the same result:
```
[2020-01-06 12:22:12.037 +01:00] [DBG] Running "FFProbeProvider" for "/media/Speelfilm/Animatie/The Secret Life of Pets 2 (2019)/The Secret Life of Pets 2.mkv"
[2020-01-06 12:22:12.038 +01:00] [DBG] "/usr/local/bin/ffprobe" "-i file:\"/media/Speelfilm/Animatie/The Secret Life of Pets 2 (2019)/The Secret Life of Pets 2.mkv\" -threads 0 -v warning -print_format json -show_streams -show_chapters -show_format"
[2020-01-06 12:22:12.038 +01:00] [DBG] Starting ffprobe with args "-i file:\"/media/Speelfilm/Animatie/The Secret Life of Pets 2 (2019)/The Secret Life of Pets 2.mkv\" -threads 0 -v warning -print_format json -show_streams -show_chapters -show_format"
[2020-01-06 12:22:13.369 +01:00] [DBG] Running "MovieNfoProvider" for "/media/Speelfilm/Animatie/The Secret Life of Pets 2 (2019)/The Secret Life of Pets 2.mkv"
[2020-01-06 12:22:13.371 +01:00] [DBG] "MovieNfoProvider" returned no metadata for "/media/Speelfilm/Animatie/The Secret Life of Pets 2 (2019)/The Secret Life of Pets 2.mkv"
[2020-01-06 12:22:13.371 +01:00] [DBG] Running "TmdbMovieProvider" for "/media/Speelfilm/Animatie/The Secret Life of Pets 2 (2019)/The Secret Life of Pets 2.mkv"
[2020-01-06 12:23:53.456 +01:00] [INF] "Scan media library" Cancelled after 3 minute(s) and 4 seconds
```
I am unable to scan any movies and if I try to manually identify it (even using the ID numbers), the identify dialogue almost immediately returns empty results.

My setup:
Version 10.4.3 (official Jellyfin docker container)
OS: Ubuntu 18.04.3
Browser: Firefox (latest), Chrome (latest)
Reverse proxy: No

All other media (home video's, music, photo's and TV shows) are scraped without issue).

**Comment by jacobmetrick** [comment]
It seems decently likely to be a linux issue based off of most of the reporters of this, it seems.

Haven't had a chance to dive into what might be going on though.

**Comment by dmsimard** [comment]
- Version 10.4.3 (Using Linuxserver [docker image](https://hub.docker.com/r/linuxserver/jellyfin))
- OS: Ubuntu 18.04.3
- Browser: Firefox (latest), Chrome (latest)
- Reverse proxy: Not relevant (reproduces with and without)

Also running into this issue -- movie libraries apparently not scanning properly and timing out with no much to go on with logs looking like this:
```
[2020-02-13 16:12:14.456 -05:00] [INF] Executing "Scan media library"
[2020-02-13 16:12:14.489 -05:00] [INF] Stopping directory watching for path "/data/movies"
[2020-02-13 16:12:14.511 -05:00] [INF] Validating media library
[2020-02-13 16:12:19.843 -05:00] [INF] MovieDbProvider: Finding id for item: Joker
```

By enabling debug logging, you get a little bit more data:
```
[2020-02-13 16:50:48.237 -05:00] [DBG] Running "FFProbeProvider" for "/data/movies/Joker.2019.MULTi.1080p.BluRay.x264-NTK.mkv"
[2020-02-13 16:50:48.264 -05:00] [DBG] "/usr/lib/jellyfin-ffmpeg/ffprobe" "-i file:\"/data/movies/Joker.2019.MULTi.1080p.BluRay.x264-NTK.mkv\" -threads 0 -v warning -print_format json -show_streams -show_chapters -show_format"
[2020-02-13 16:50:48.265 -05:00] [DBG] Starting ffprobe with args "-i file:\"/data/movies/Joker.2019.MULTi.1080p.BluRay.x264-NTK.mkv\" -threads 0 -v warning -print_format json -show_streams -show_chapters -show_format"
[2020-02-13 16:50:48.768 -05:00] [DBG] Running "MovieNfoProvider" for "/data/movies/Joker.2019.MULTi.1080p.BluRay.x264-NTK.mkv"
[2020-02-13 16:50:48.781 -05:00] [DBG] "MovieNfoProvider" returned no metadata for "/data/movies/Joker.2019.MULTi.1080p.BluRay.x264-NTK.mkv"
[2020-02-13 16:50:48.782 -05:00] [DBG] Running "TmdbMovieProvider" for "/data/movies/Joker.2019.MULTi.1080p.BluRay.x264-NTK.mkv"
[2020-02-13 16:50:52.941 -05:00] [INF] MovieDbProvider: Finding id for item: Joker
[2020-02-13 16:50:53.294 -05:00] [DBG] Running "OmdbItemProvider" for "/data/movies/Joker.2019.MULTi.1080p.BluRay.x264-NTK.mkv"
```

Seeing that ``OmdbItemProvider`` was the last thing that ran, I tried disabling the ``The Open Movie Database`` through the interface like this:

![Screenshot from 2020-02-13 17-10-24](https://user-images.githubusercontent.com/1291204/74483128-cae66300-4e83-11ea-9bad-99c53208f1f6.png)

Once I disabled the OMDB provider and re-triggered a library scan, everything unblocked rather quickly with the logs looking like this instead:
```
[2020-02-13 16:52:17.512 -05:00] [DBG] Running "FFProbeProvider" for "/data/movies/Joker.2019.MULTi.1080p.BluRay.x264-NTK.mkv"
[2020-02-13 16:52:17.514 -05:00] [DBG] "/usr/lib/jellyfin-ffmpeg/ffprobe" "-i file:\"/data/movies/Joker.2019.MULTi.1080p.BluRay.x264-NTK.mkv\" -threads 0 -v warning -print_format json -show_streams -show_chapters -show_format"
[2020-02-13 16:52:17.515 -05:00] [DBG] Starting ffprobe with args "-i file:\"/data/movies/Joker.2019.MULTi.1080p.BluRay.x264-NTK.mkv\" -threads 0 -v warning -print_format json -show_streams -show_chapters -show_format"
[2020-02-13 16:52:17.619 -05:00] [DBG] Running "MovieNfoProvider" for "/data/movies/Joker.2019.MULTi.1080p.BluRay.x264-NTK.mkv"
[2020-02-13 16:52:17.619 -05:00] [DBG] "MovieNfoProvider" returned no metadata for "/data/movies/Joker.2019.MULTi.1080p.BluRay.x264-NTK.mkv"
[2020-02-13 16:52:17.619 -05:00] [DBG] Running "TmdbMovieProvider" for "/data/movies/Joker.2019.MULTi.1080p.BluRay.x264-NTK.mkv"
[2020-02-13 16:52:17.619 -05:00] [INF] MovieDbProvider: Finding id for item: Joker
[2020-02-13 16:52:17.680 -05:00] [DBG] Running "TmdbImageProvider" for "/data/movies/Joker.2019.MULTi.1080p.BluRay.x264-NTK.mkv"
[2020-02-13 16:52:17.725 -05:00] [DBG] Throttling Tmdb by 193 ms
```

So it looks like it is the OMDB library metadata downloader that is not behaving properly and causing the library scan to fail or timeout ?

**Comment by nygage** [comment]
i have the same error as @dmsimard disabled TOMD and it seems to work

**Comment by themoken** [comment]
I had this issue on 10.5.5 on Arch but it affected both OMDB and TMDB so I looked harder. It looks like this might be another ipv6 issue (like #2350 and related to #1002) which may help explain why this only happens on Linux. Doing anything with omdbapi.com (pinging, going via browser etc.) doesn't appear to work over ipv6 so ultimately I think this is on them for having non-working ipv6 addresses in DNS, but it would be nice for Jellyfin to fallback on ipv4 instead of failing.

One can fix this by disabling ipv6 or, as a workaround, I added this to `/etc/hosts`

<pre>
104.20.134.15   omdbapi.com www.omdbapi.com img.omdbapi.com
13.249.71.53    themoviedb.org www.themoviedb.org
13.225.53.99    api.themoviedb.org
</pre>

Now scanning works again with lookups from both databases.

**Comment by tdehoog** [comment]
I had this issue too, which kept me from using Jellyfin and staying with Plex... Based on @themoken 's comment I decided to create a new bridge network, but without specifying any IPv6 details so it creates an IPv4 only network in my docker environment. I connected my Jellyfin container to this new network and voila, it magically started indexing my library...

I would like to suggest making the scraping of metadata more robust.

**Comment by BeyhanSen** [comment]
@themoken's solution worked for me as well. First I attempted to start with Jellyfin a few months ago but when I couldn't figure out I ended up using Plex. Hopefully I will continue on Jellyfin. If you can solve this sort of maybe small but bad user experience bugs I think many will start using Jellyfin.

**Comment by Shadowghost** [comment]
It's not a problem caused by jellyfin itself. The metadata providers do have IPv6 connectivity (AAAA DNS records) but do not server (all) API endpoints on them. By overriding the DNS -> IP resolution to use IPv4 addresses for the respective domains (done by the host file entries above) you circumvent this problem.
Hardcoding those IPs in the code would be really bad practice. It should be mentioned in the docs though.

**Comment by marekyggdrasil** [comment]
The issue persists. I just moved my library and setup correct permissions, nothing appears on my Jellyfin and it is most likely because metadata cannot be fetched.

**Comment by marekyggdrasil** [comment]
I encountered this problem and just solved it.

In my case, I had a library `Music` pointed to `/var/lib/transmission-daemon/downloads` and later I decided to split it into three libraries based on the content pointing to following directories.

Music -> `/var/lib/transmission-daemon/downloads/music`
Movies -> `/var/lib/transmission-daemon/downloads/movies`
Series -> `/var/lib/transmission-daemon/downloads/series`

The `Music` was same library as before, I just removed previous folder and added new one in the settings.

I created the directories and moved the content, ensured proper ownership and access rights, but I still couldn't see any content on my Jellyfin.

I examined the logs and found this:

```
[2020-12-08 09:07:02.348 +00:00] [INF] Watching directory /var/lib/transmission-daemon/downloads
```

and the following keep repeating

```
[2020-12-08 09:04:00.308 +00:00] [INF] Found duplicate path: "/var/lib/transmission-daemon/downloads/movies"
[2020-12-08 09:04:00.308 +00:00] [INF] Found duplicate path: "/var/lib/transmission-daemon/downloads/series"
```

It turned out that even thou I edited the `Music` library to point into `/var/lib/transmission-daemon/downloads/music` instead of `/var/lib/transmission-daemon/downloads` it kept checking the latter and it was caused conflicts. Solution was to remove the libraries and add them again pointing to appropriate directories. Then there was no longer any conflicts and everything synced properly.
